From e8e84b725b2d9048f34c57c85893e4c35d94ea9e Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 31 Jul 2019 17:39:37 +0100
Subject: [PATCH 735/826] overlays: Add audio parameter to vc4-kms-v3d

The audio parameter to the vc4-kms-v3d overlay allows audio support
to be disabled (it defaults to on) by adding "audio=off" to the
dtoverlay parameters.

See: https://github.com/raspberrypi/linux/issues/2489

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/README                  | 1 +
 arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 203041654d84..6131ac3d967c 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -2480,6 +2480,7 @@ Params: cma-256                 CMA is 256MB (needs 1GB)
         cma-128                 CMA is 128MB
         cma-96                  CMA is 96MB
         cma-64                  CMA is 64MB
+        audio                   Enable or disable audio over HDMI (default "on")
 
 
 Name:   vga666
diff --git a/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts b/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
index 19e1d2548e7b..c5f687e8bcb9 100644
--- a/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
@@ -134,11 +134,19 @@
 		};
 	};
 
+	fragment@17 {
+		target = <&hdmi>;
+		__dormant__  {
+			dmas;
+		};
+	};
+
 	__overrides__ {
 		cma-256 = <0>,"+0-1-2-3-4";
 		cma-192 = <0>,"-0+1-2-3-4";
 		cma-128 = <0>,"-0-1+2-3-4";
 		cma-96  = <0>,"-0-1-2+3-4";
 		cma-64  = <0>,"-0-1-2-3+4";
+		audio   = <0>,"!17";
 	};
 };
-- 
2.22.0

